src/main/java/duke/Bobi.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import duke.exception.EmptyCommandException;
+import duke.exception.InvalidCommandException;
+import duke.utility.Command;
+import duke.utility.Storage;
+import duke.utility.TaskList;
+import duke.utility.Ui;
+
+/**
+ * Bobi class encapsulates all components of the Bobi tracker bot.
+ *
+ * @author ruo-x
+ */
+public class Bobi {
+    private Storage storage;
+    private TaskList list;
+    private Ui ui;
+    private Command command;
+
+    /**
+     * Constructor of a Bobi object.
+     * Load tasks saved from the backend into the active Bobi.
+     */
+    public Bobi() {
+        this.storage = new Storage();
+        this.list = new TaskList();
+
+        // Load list of tasks stored in text file "task.txt" into the local task List
+        storage.handleLoad(list);
+
+        // Start Scanner to read user inputs
+        this.ui = new Ui(list);
+        this.command = new Command(storage, ui, list);
+    }
+
+    /**
+     * Runs Bobi.
+     * Directs Bobi to perform certain actions based on the user's input.
+     */
+    public void run() {
+        System.out.println(Ui.greeting());
+
+        String answer = ui.getInput();
+
+        // Listens to user commands and perform the relevant functions
+        while (true) {
```

Comment:
Understandable that this is a really long chunk as theres really alot of type of input, could be simpler to divide the commands into creating tasks and handling tasks to shorten this!